src/mozilla_taskgraph/worker_types.py

```diff
@@ -17,9 +17,32 @@
                 "env", description="Environment variables to pass into the build"
             ): {str: taskref_or_string},
             Optional(
-                "build_params",
-                description="Parameters describing the build context to pass "
-                "onto Bitrise. All keys are optional but specific workflows "
+                "workflow_params",
+                description="Workflow-specific parameters describing the build context to pass "
+                "onto Bitrise. The key should be the workflow id, and mapped to the params to be added. "
+                "Overrides global_params if there's a conflict. Lists are MERGED.",
```

Comment:
It's unfortunate that tasks would need to specify the `workflow` both in `workflows` and in `workflow_params`. I'm trying to think of a better way of achieving this..

Keep in mind that this schema doesn't necessarily need to map directly to the bitrisescript one. We can make things a little easier for task authors here by providing a translation layer.

Reply:
I'm down for making this easier to use